### PR TITLE
Fix install layout db options

### PR DESCRIPTION
**Pull request by craigpanton**
## Description
Adding the Advanced DB options to the original More Options list and fixing layout issues with the install page.

## Motivation and Context
It improves the layout of the install page.

## How To Test This
Go through the installation process, ensure that all install pages display correctly (specifically that the white border round the text covers the full list of options). Ensure that you can install as expected.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: install/installConfig.php [hunk 1]
```diff
@@ -1258,10 +1192,58 @@ private function getFormItems(
...
 </div>
 EOQ;
+        $out .= "</div>";
+
+        // --------------------------
+        //  Advanced Database Configuration
+        // --------------------------------->
 
-        $out.= "</div>";
+        require_once('install/suite_install/collations.php');
```
Comment (Dillon-Brown) [inline install/installConfig.php]: This should use``` __DIR__``` to guarantee we load the right file.

File: install/installConfig.php [hunk 2]
```diff
@@ -1258,10 +1192,58 @@ private function getFormItems(
...
 
+        $collationCB = "<select name='setup_db_collation' id='setup_db_collation' class='select' onChange='document.getElementById(\"setup_db_charset\").value = document.getElementById(\"setup_db_collation\").value.split(\"_\")[0];'>";
```
Comment (Dillon-Brown) [inline install/installConfig.php]: Ideally, all of this should be moved into a template but this should be fine for a quick fix.

File: install/installConfig.php [hunk 3]
```diff
@@ -1258,10 +1192,58 @@ private function getFormItems(
...
+        $charset = "<select name='setup_db_charset' id='setup_db_charset' class='select'>";
+
+        if (isset($collations)) {
```
Comment (Dillon-Brown) [inline install/installConfig.php]: ```$collations``` isn't defined in the scope.

File: install/installConfig.php [hunk 4]
```diff
@@ -1258,10 +1192,58 @@ private function getFormItems(
...
+            if (isset($_SESSION['setup_db_type']) && $_SESSION['setup_db_type'] == "mysql") {
```
Comment (Dillon-Brown) [inline install/installConfig.php]: This statement can be merged with the parent. I would also check to see if we can use strict comparison here.

File: install/installConfig.php [hunk 5]
```diff
@@ -1258,10 +1192,58 @@ private function getFormItems(
...
+                foreach ($collations['mysql'] as $collation) {
+                    $collationCB .= "<option value='" . $collation['name'] . "' >" . $collation['name'] . "</option>";
+                    $charset .= "<option value='" . $collation['charset'] . "' >" . $collation['charset'] . "</option>";
+                }
+            }
+        }
+
+        $collationCB .= "</select>";
```
Comment (Dillon-Brown) [inline install/installConfig.php]:
```suggestion
        $collationCB .= '</select>';
```

File: install/installConfig.php [hunk 6]
```diff
@@ -1258,10 +1192,58 @@ private function getFormItems(
...
+        $charset .= "</select>";
```
Comment (Dillon-Brown) [inline install/installConfig.php]:
```suggestion
        $charset .= '</select>';
```

File: install/installConfig.php (hunk 3 above)
Comment (craigpanton) [reply install/installConfig.php]: $collations is defined within the collations.php requirement at line 1201.

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ install/installConfig.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/samus-aran/SuiteCRM/pullRequest?prid=4831833&bid=16052131)

**Comment by Mac-Rae** [comment]
Assessed :+1:

**Pull request merged**